content/docs/other-guides/istio-in-kubeflow.md

```diff
@@ -53,9 +52,45 @@ Kubeflow uses Istio as a uniform way to secure, connect, and monitor microservic
 
 ## Istio in Kubeflow
 
-TODO (abhishek): Elaborate further...
+The following diagram illustrates how user requests interact with services in
+Kubeflow. It walks through the process when a user requests to create a new
+notebook server via the Notebooks Servers UI accessible through the Central Dashboard.
+
 
 <img src="/docs/images/Istio-in-KF.svg" 
   alt="Select active profile "
   class="mt-3 mb-3 border border-info rounded">
 
+  1. The user request is intercepted by an identification proxy which talks to
+     a SSO service provider such as IAM on Cloud Services Provider or Active
+     Directory/LDAP on-premises. 
+  1. Once the user is authenticated, the request is modified by the Istio
+     Gateway to include a JWT Header token containing the identity of the user.
+     All requests thorughout the service mesh carry this token along.
+  1. The Istio RBAC policies are applied on the incoming request to validate
+     the access to the service and the requested namespace. If either of those
+     are inaccessible to the user, an error response is sent back. 
+  1. If the request is validated, it is forwarded to the appropriate controller
+     (Notebooks Controller in this case).
+  1. Notebooks Controller will create the notebook pod in the namespace user requested.
+
+Further actions by the user with the notebook to create training jobs or other
+resources in the namespace go through a similar process. Profiles Controller
+manages the creation of profiles, creates and applies appropriate Istio
+policies. For more details, please see [multi-user
+overview](/docs/other-guides/multi-user-overview/).
+
+
+## Deploying Kubeflow without Isito
+
+It is not possible to deploy Kubeflow without Istio. Kubeflow needs the Istio
+CRDs to express the new route to access the created Notebook from the Gateway.
```

Comment:
Hello @abhi-g . 
I like the section `Kubeflow needs Istio`, but think the heading could be reworded.

Lines 41-42, contain a lot of jargon. Do you need `middleware setup`?
`Kubeflow provides the underlying infrastructure and middleware setup that makes
it possible to be able to put such disparate components together.`

You might want to mention (in that section) that besides being a Service Mesh, Istio is a Custom Resource Definition?
I like this description of the role of Istio:
`Kubeflow uses Istio as a uniform way to secure, connect, and monitor microservices`

Is it important to provide suggestions for alternative ways of routing (`Deploying Kubeflow without Istio` section) if this is not possible today? If relevant, could you add a link to the Service Mesh Interface doc or proposal?
Clarify `the Gateway` (Istio gateway)?
How do you manage Istio with regard to Kubeflow (read the Istio docs)?

Reply:
Thanks @kbhawkey. Made updates based on your comments. Removed any information that was unnecessary or not actionable for now. I didn't add the part about Istio also being a CRD because it is itself a complex system with many CRDs.